Common/Net/Dns/CachedIPHostEntry.cs

```diff
@@ -23,7 +22,7 @@ internal static CachedIPHostEntry CreateErrorEntry(IPAddress ip, Exception e)
                 HostEntry = new IPHostEntry
                 {
                     HostName = "unknown",
-                    AddressList = ip != null ? new IPAddress[] { ip } : Array.Empty<IPAddress>()
+                    AddressList = ip != IPAddress.None ? new[] { ip } : Array.Empty<IPAddress>()
```

Comment:
Nope, IPAddress.None is used to avoid using a null pattern when we can. Like int.MinValue to check if an int has been set to a specific value or not, for instance...
It's semantically better to use predefined "unknown" / "no-value" constants than null when possible, IMHO.